ztp/siteconfig-generator/siteConfig/siteConfig.go

```diff
@@ -179,6 +179,13 @@ type Clusters struct {
 	NumWorkers        uint8
 	ClusterType       string
 	CrTemplates       map[string]string `yaml:"crTemplates"`
+	LegacyConfig      *LegacyConfig     `yaml:"legacyConfig"`
+}
+
+// LegacyConfig set of configurations to assist with backwards compatibility
+type LegacyConfig struct {
```

Comment:
Instead of identifying this as legacy vs current maybe just a field like `mergeDefaultMachineConfigs`?